activities/Chess.activity/js/chessGame.js

```diff
@@ -391,18 +391,24 @@ var ChessGame = {
       var start_time = Date.now();
       var possibleMove = this.state.findmove(this.level);
       var delta = Date.now() - start_time;
+
+      // console.log(`Random Move: ${possibleMove}, Delta: ${delta}`);
+
       // game over
-      if (possibleMove.length === 0) {
+      if (possibleMove.length === 0 && (move.flags === 7 || move.flags === 15 || move.flags === 23 || move.flags === 39)) {
+        // console.log("Game won by computer");
         this.game_won = true;
         return;
       }
-      if (1 / possibleMove[2] == 1 / this.state.stalemate_scores[this.state.to_play]) {
+  
+      if (possibleMove.length === 0 && (move.flags === 5 || move.flags === 13 || move.flags === 21 || move.flags === 37 || move.flags === 64)) {
+        // console.log("Match draw due to stalemate");
         this.game_draw = true;
         return;
       }
       var move = this.state.move(possibleMove[0], possibleMove[1]);
 
-      if (!(move.flags & (1 << 0))) {
+      if (!(move.flags & (1))) {
```

Comment:
This line do the same test than before

Reply:
I kept numericals because from the library it's easy to read via numbers because they have assigned number to certain things in chess. I could change that if you would say

Reply:
Yes please. Don't change what doesn't need changing.

Reply:
I have changed it as it was

Reply:
And made the appropriate logic changes to solve the issue.